examples/helloworld/greeter_server/main.go

```diff
@@ -55,11 +60,25 @@ func (s *server) SayHello(ctx context.Context, in *pb.HelloRequest) (*pb.HelloRe
 }
 
 func main() {
-	lis, err := net.Listen("tcp", port)
-	if err != nil {
-		log.Fatalf("failed to listen: %v", err)
-	}
+	flag.Parse()
+
 	s := grpc.NewServer()
 	pb.RegisterGreeterServer(s, &server{})
-	s.Serve(lis)
+
+	log.Printf("Running hello server on %s ...", *listen)
+	if *httpMode {
+		// Running a gRPC server when you need to integrate with an existing
+		// net/http server on the same port: (using HTTP routing as needed)
+		http.Handle("/", s)
```

Comment:
Found it. We are using a [custom `net.Listener` implementation](https://github.com/cockroachdb/cockroach/blob/gossip-grpc/util/net.go#L34:L101), and hitting this: https://github.com/golang/go/blob/release-branch.go1.5/src/net/http/server.go#L1296

See also https://github.com/golang/go/issues/12737.